testing/apptest-gc/manifests.yaml

```diff
@@ -0,0 +1,84 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: apptest-namespaces-gc
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: namespace-edit
+rules:
+- apiGroups: [""]
+  resources: ["namespaces"]
+  verbs: ["*"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: apptest-namespaces-gc-crb
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: namespace-edit
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: apptest-namespaces-gc
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: gc-script
+  namespace: apptest-namespaces-gc
+data:
+  gc.sh: |-
+    set -eo pipefail
+
+    namespaces=($(kubectl get namespaces -o json \
+      | jq -r \
+        '.items[]
+         | { "name": .metadata.name, "time": (now - (.metadata.creationTimestamp|fromdate)) }
+         | select(.name | test("^apptest-\\w{8,16}$"))
+         | select(.time >= 1800)
+         | .name'))
+
+    echo "Number of eligible namespaces: ${#namespaces[@]}"
+
+    DRYRUN=${DRYRUN:-true}
```

Comment:
Looks like DRYRUN defaults to true - does this mean that the Cron is disabled by default?

Reply:
Oops, meant to set an environment variable for the container spec.